src/uct/ib/base/ib_md.h

```diff
@@ -504,10 +504,17 @@ int uct_ib_device_is_accessible(struct ibv_device *device);
 
 ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
                                    struct ibv_device *ib_device,
-                                   const uct_ib_md_config_t *md_config);
+                                   const uct_ib_md_config_t *md_config,
+                                   size_t mem_size, size_t mr_size);
+
+void uct_ib_md_close_common(uct_ib_md_t *md);
+
+void uct_ib_md_cleanup_common(uct_ib_md_t *md);
 
 void uct_ib_md_close(uct_md_h uct_md);
 
+void uct_ib_md_cleanup(uct_ib_md_t *md);
```

Comment:
this creates more mess - what is the difference between close_common, cleanup_common, close, cleanup?
are there also 4 different "init/open" functions to match these 4 "close/clenaup" functions?

Reply:
refactored